sources/api/host-containers/src/main.rs

```diff
@@ -437,8 +437,32 @@ where
     Ok(())
 }
 
+fn is_container_affected(settings: &[&str], container_name: &str) -> bool {
+    if settings.is_empty() {
+        // it means that Bottlerocket is booting - all containers need to be started
+        return true;
+    }
+    let setting_prefix = "settings.host-containers.";
+    let container_prefix = format!("{}{}.", setting_prefix, container_name);
+
+    for setting in settings {
+        if setting.starts_with(&container_prefix) {
+            info!("Found a changed setting for {} container", container_name);
```

Comment:
I didn't write it for the settings that affects all host-containers since it would print that  `number of container` of times

Reply:
That seems fine - it seems like it'd be useful to know why a container is being handled.  The user might not have changed any settings with that container in the name, for example.  They might not realize that network settings affect it.  Or that sometimes we have to handle all containers.  Or maybe they expected a container to be affected and it wasn't, so clarifying that could help.  When debugging, consistent output is good :)

Reply:
I think it'd be helpful for all four return locations for consistency.  It'd also help to include the setting name that triggered the change in the log messages.

Reply:
okay, about the setting name, I was thinking if there are multiple settings for same container, only first one would be printed since we are returning after a match, thus I didn't add it

Reply:
We could just include the first, if the messaging was clear, or maybe instead of returning true/false immediately we could build a list of settings that were relevant and include them in the log...  or leave it as-is :)